tests/helpers/perf_run_helper.rb

```diff
@@ -344,10 +411,12 @@ def handle_scale_results(scenario_hash)
     copy_scale_results(scenario)
 
     # check results for KOs
-    check_scale_results(scenario_hash)
+    success = check_scale_results(scenario_hash)
 
     # perform assertions
-    scale_assertions(atop_result, gatling_result)
+    scale_assertions(atop_result, gatling_result) unless !success
+
+    return success
```

Comment:
This appears to return the result of check_scale_results.  But when I read a method like scale_assertions I would expect it might find things that are relevant to the return value...

Reply:
I think the idea here was to not bother performing the assertions if check_scale_results has already determined that the iteration was not successful. I was having issues getting the results cleanly at the end if the Beaker run failed, but perhaps that should be handled using the teardown.

Reply:
ok